Formula/git-filter-repo.rb

```diff
@@ -0,0 +1,35 @@
+class GitFilterRepo < Formula
+  desc "Quickly rewrite git repository history"
+  homepage "https://github.com/newren/git-filter-repo"
+  url "https://github.com/newren/git-filter-repo/releases/download/v2.24.0/git-filter-repo-2.24.0.tar.xz"
+  sha256 "92188d3c44b9ff0dd40dfeed72859e0a088f775c12fb24c4e3e27a8064cfcc84"
+
+  # ignore git dependency audit:
+  #  * Don't use git as a dependency (it's always available)
+  # https://github.com/Homebrew/homebrew-core/pull/46550#issuecomment-563229479
+  depends_on "git"
+  depends_on "python"
```

Comment:
Does this work with the macOS system Python on any versions?

Reply:
According to the project itself, it should work with python3. no version mentioned, so I guess any version will do.
- https://github.com/newren/git-filter-repo#how-do-i-install-it

as the shebang uses `env`, any available binary with `python3` name will be used:
```
 ➔ head -n1 /usr/local/bin/git-filter-repo
#!/usr/bin/env python3
```

Reply:
In that case I think a 
```ruby
if MacOS.version >= :catalina 
  uses_from_macos "python3"
else
  depends_on "python3"
end
```

would be suitable

Reply:
Added exactly like that in c4a02573c1